src/components/__tests__/CSSTable.test.tsx

```diff
@@ -0,0 +1,137 @@
+        value: '#123456',
+      },
+    },
+    '.another-selector': {
+      '--list-property': {
+        name: 'list-property',
+        value: 'list-item-1',
+        values: ['list-item-1', 'list-item-2'],
+      },
+    },
+  },
+  'pf-v6-empty': {},
+}))
+
+describe('<CSSTable />', () => {
+  it('renders without crashing', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" />)
+    expect(
+      screen.getByRole('grid', {
+        name: /CSS Variables prefixed with pf-v6-test/i,
+      }),
+    ).toBeInTheDocument()
+  })
+
+  it('renders the correct table headers when hideSelectorColumn is false (default)', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" />)
+    expect(screen.getByText('Selector')).toBeInTheDocument()
+    expect(screen.getByText('Variable')).toBeInTheDocument()
+    expect(screen.getByText('Value')).toBeInTheDocument()
+  })
+
+  it('renders the correct table headers when hideSelectorColumn is true', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" hideSelectorColumn />)
+    expect(screen.queryByText('Selector')).not.toBeInTheDocument()
+    expect(screen.getByText('Variable')).toBeInTheDocument()
+    expect(screen.getByText('Value')).toBeInTheDocument()
+  })
+
+  it('renders the correct data rows', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" />)
+    expect(
+      screen.getByRole('row', {
+        name: '.test-selector test-property test-value',
+      }),
+    ).toBeInTheDocument()
+    expect(
+      screen.getByRole('row', {
+        name: '.test-selector another-property (In light theme) #123456',
+      }),
+    ).toBeInTheDocument()
+    expect(
+      screen.getByRole('row', {
+        name: 'Details .another-selector list-property list-item-1',
+      }),
+    ).toBeInTheDocument()
+  })
+
+  it('renders the "Prefixed with" header when autoLinkHeader is true', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" autoLinkHeader />)
+    expect(
+      screen.getByRole('heading', {
+        name: /Prefixed with 'pf-v6-test'/i,
+        level: 3,
+      }),
+    ).toBeInTheDocument()
+  })
+
+  it('does not render the "Prefixed with" header when autoLinkHeader is false (default)', () => {
+    render(<CSSTable cssPrefix="pf-v6-test" />)
+    expect(
+      screen.queryByRole('heading', {
+        name: /Prefixed with 'pf-v6-test'/i,
+        level: 3,
+      }),
+    ).not.toBeInTheDocument()
+  })
+
+  it('filters rows based on search input', async () => {
+    render(<CSSTable cssPrefix="pf-v6-test" debounceLength={0} />)
+    const searchInput = screen.getByPlaceholderText('Filter CSS Variables')
+
+    expect(
+      screen.getByRole('row', {
+        name: '.test-selector test-property test-value',
+      }),
+    ).toBeInTheDocument()
+    expect(
+      screen.getByRole('row', {
+        name: 'Details .another-selector list-property list-item-1',
+      }),
+    ).toBeInTheDocument()
+    fireEvent.change(searchInput, { target: { value: 'test' } })
```

Comment:
Nit: I would use userEvent rather than fireEvent, unless for some reason userEvent doesn't work here.